sycl/include/sycl/ext/intel/experimental/task_sequence.hpp

```diff
@@ -0,0 +1,136 @@
+/* INTEL_CUSTOMIZATION */
+//==--------------------------- task_sequence.hpp --------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/__spirv/spirv_ops.hpp>
+#include <CL/__spirv/spirv_types.hpp>
+#include <sycl/exception.hpp>
+#include <sycl/ext/intel/experimental/fpga_kernel_properties.hpp>
+#include <sycl/ext/intel/experimental/task_sequence_properties.hpp>
+#include <sycl/ext/oneapi/properties/properties.hpp>
+
+#include <type_traits>
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::intel::experimental {
+
+template <auto &f> struct task_sequence_checker : std::false_type {};
+
+template <typename ReturnT, typename... ArgsT, ReturnT (&f)(ArgsT...)>
+struct task_sequence_checker<f> : std::true_type {};
+
+template <auto &f,
+          typename PropertyListT = oneapi::experimental::empty_properties_t>
+class task_sequence {
+
+  static_assert(task_sequence_checker<f>::value);
+
+  static_assert(oneapi::experimental::is_property_list<PropertyListT>::value,
+                "Property list is invalid.");
+};
+
+template <typename ReturnT, typename... ArgsT, ReturnT (&f)(ArgsT...),
+          typename... Props>
+#if defined(__SYCL_DEVICE_ONLY__)
+class
+    [[__sycl_detail__::__uses_aspects__(aspect::ext_intel_fpga_task_sequence)]] task_sequence<
+#else
+class task_sequence<
+#endif
+        f, oneapi::experimental::detail::properties_t<Props...>> {
+  using property_list_t = oneapi::experimental::detail::properties_t<Props...>;
+
+public:
+  task_sequence(const task_sequence &) = delete;
+  task_sequence &operator=(const task_sequence &) = delete;
+  task_sequence(task_sequence &&) = delete;
+  task_sequence &operator=(task_sequence &&) = delete;
+
+  task_sequence() {
+#if defined(__SYCL_DEVICE_ONLY__)
+    taskSequence = __spirv_TaskSequenceCreateINTEL(
+        &f, pipelined, fpga_cluster, response_capacity, invocation_capacity);
+#else
+    throw exception{make_error_code(errc::feature_not_supported),
+                    "task_sequence is not supported on host device"};
```

Comment:
```suggestion
                    "task_sequence is not supported on the host"};
```
since host device has been dropped. The same goes for the two functions below.

Reply:
changed